### ISSUE TITLE
Frameless Window

### ISSUE THREAD

**Issue by khalidchawtany** [neovide/neovide#67]
Can we have frameless window support.

I am using a modified version of [neovim-qt](https://github.com/khalidchawtany/neovim-qt/commit/9f2af46da7c4013d7cecce68db146ee8ef047bce) in which I remove the window frame to make it clean. 
It is draggable resizable  and everything. 

![2020-01-28 09 45 00](https://user-images.githubusercontent.com/1681606/73241465-1f66be80-41b3-11ea-937a-35e49f527a68.gif)

**Comment by Kethku** [comment]
Oooh that looks sick. I'll see if I can make it happen

**Comment by Kethku** [comment]
Its not looking good. Winit no decorations disables window resizing.

**Comment by Kethku** [comment]
I did some fiddling and I think I could emulate resizing and window movement, but getting it to feel exactly right is going to be hard. For example on windows, window key plus arrow key should move it automatically which will be hard to do exactly right...

**Comment by Kethku** [comment]
Heres the branch of what I have so far. https://github.com/Kethku/neovide/tree/testing-frameless

Any chance someone can try this branch and see if resize works on other platforms? This issue mentions the windows platform issue, but doesn't say if linux and mac work: https://github.com/rust-windowing/winit/issues/725

**Comment by khalidchawtany** [comment]
I will try it when I get back home and report back. Thanks

**Comment by khalidchawtany** [comment]
I get a panic running the specified branch.
```
$ RUST_BACKTRACE=1 ./neovide
thread 'main' panicked at 'Failed to parse emoji font data: Parse', src/libcore/result.rs:1165:5
stack backtrace:
   0: std::io::Read::read_exact
   1: core::fmt::ArgumentV1::show_usize
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
   6: std::panicking::continue_panic_fmt
   7: std::panicking::try::do_call
   8: std::thread::local::fast::Key<T>::try_initialize
   9: std::thread::local::fast::Key<T>::try_initialize
  10: core::result::Result<T,E>::expect
             at /private/tmp/rust-20191221-83947-1gnlhtt/rustc-1.40.0-src/src/libcore/result.rs:960
  11: neovide::renderer::caching_shaper::build_collection_by_font_name
             at src/renderer/caching_shaper.rs:88
  12: neovide::renderer::caching_shaper::FontSet::new
             at src/renderer/caching_shaper.rs:110
  13: neovide::renderer::caching_shaper::CachingShaper::new
             at src/renderer/caching_shaper.rs:149
  14: neovide::renderer::Renderer::new
             at src/renderer/mod.rs:30
  15: neovide::window::ui_loop
             at src/window.rs:33
  16: std::rt::lang_start::{{closure}}
             at /private/tmp/rust-20191221-83947-1gnlhtt/rustc-1.40.0-src/src/libstd/rt.rs:61
  17: std::panicking::try::do_call
  18: panic_unwind::imp::find_eh_action::{{closure}}
  19: std::sys_common::cleanup
  20: neovide::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

```

**Comment by Kethku** [comment]
Ah this was my bad. I will rebase on master where this bug was fixed

**Comment by Kethku** [comment]
Ok try now on that branch. I rebased and force pushed

**Comment by khalidchawtany** [comment]
Everything works nicely on MacOS except for the part that I cannot drag the window around. In the patched version of neovim-qt I can drag the window using the top region of the window. The region where the top bar would have appeared if not frameless.

**Comment by Kethku** [comment]
Okeydokey. That shouldn't be too hard and I could even merge a version of this for mac and linux where it actually works and then merge windows support later. Dragging the window around shouldn't be too hard. This is encouraging :)

**Comment by Kethku** [comment]
Question: How important do you think the rounded corners are?

**Comment by khalidchawtany** [comment]
It is actually a nice addition but I don't think it is  bummer.

**Comment by Kethku** [comment]
blocked on https://github.com/Kethku/neovide/issues/97

**Comment by khalidchawtany** [comment]
Hi @Kethku,
Since https://github.com/Kethku/neovide/issues/97 is close what is the status of this?
No pressure I was just asking :)

**Comment by mborejdo** [comment]
also chiming in and looking forward to this.

**Comment by i3d** [comment]
on Mac, iTerm2 uses a technique to embed the frame with the terminal window. This works fairly well.

yes, frameless would make neovide way more slick.

![Screen Shot 2021-04-14 at 14 42 07](https://user-images.githubusercontent.com/105068/114784124-a14af500-9d2f-11eb-818c-9d262f71b2d7.png)

**Comment by Kethku** [comment]
Some more requests for this have arrived. I think this is blocked on upstream winit support as toolbar-less windows are not resizable. I am open to PRs for this though if anybody wants to dig in

**Comment by shaunsingh** [comment]
Chiming in to say it would be nice to have this even without window resizing. For those of us with tiling window managers it takes care of it. Doom emacs does this without resizing and rounded corners and its comes out quite nice 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196912/118757088-5181b100-b83a-11eb-9c16-080c217a87a5.png">

**Comment by Kethku** [comment]
Thats a good point. I'll look into what it would entail.

**Comment by shaunsingh** [comment]
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196912/120224181-ee682500-c210-11eb-9b2d-d0b906fe2f40.png">

works well on macOS, thanks!

**Comment by shaunsingh** [comment]
One issue I've run into is that a frameless window can't be moved to a fullscreen space: 
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196912/120900810-04a52500-c605-11eb-80e4-53c53a468248.png">

Frameless emacs and iterm seem to work fine.

**Comment by shaunsingh** [comment]
This might just be an issue with rust/something upstream and can't be fixed with neovide. alacritty has the same behavior

**Comment by PyGamer0** [comment]
@shaunsingh probably because the window is not resizeable

**Comment by Sayan-Manna** [comment]
> <img alt="image" width="1440" src="https://user-images.githubusercontent.com/71196912/120224181-ee682500-c210-11eb-9b2d-d0b906fe2f40.png">
> 
> works well on macOS, thanks!

Which WM you're using? And how r u making it fool screen using keyboard?

**Comment by shaunsingh** [comment]
> Which WM you're using?
I'm using yabai
> And how r u making it fool screen using keyboard?
Both windows are still in windowed mode. The PR implemented a while back makes it frameless, i.e. removes the corners and window decorations

**Comment by chardskarth** [comment]
@shaunsingh how do you start neovide? 
I already have in my `.zprofile`

```
export NEOVIDE_FRAMELESS=true
export NEOVIDE_FRAME=none
```

Running in the terminal: `neovide --frame none` fixes it for me. 
But i need it to open frameless when called in Mac's Spotlight.

**Comment by fathergoose** [comment]
> @shaunsingh how do you start neovide? I already have in my `.zprofile`
> 
> ```
> export NEOVIDE_FRAMELESS=true
> export NEOVIDE_FRAME=none
> ```
> 
> Running in the terminal: `neovide --frame none` fixes it for me. But I need it to open frameless when called in Mac's Spotlight.
@chardskarth 
For MacOS gui apps you need to put the values into `/etc/launchd.conf`. Its covered in the answer to this stack overflow question. https://stackoverflow.com/questions/135688/setting-environment-variables-on-os-x

**Comment by sinnrrr** [comment]
That still doesn't work with the window manager of choice. Attaching a video of the current behavior here.

https://user-images.githubusercontent.com/25615887/172051186-84c10820-4e10-402c-bd9b-c36fecb09d8f.mov

**Comment by Matts966** [comment]
@Kethku 
Thank you for this great project :)
I am interested in this issue and made a pilot PR in #1459.
I am also interested in continuously maintaining this change.
Could you kindly review or introduce another maintainer?
Thank you 🙇 

Also, I would like to ask people here whether this PR can meet our requirements.

https://user-images.githubusercontent.com/28551465/183240179-b7d15744-3d9d-47c6-8008-30a5a250ae7b.mov

**Comment by timmyha** [comment]
> @ankushKun better, now it can be moved with the cursor, but still doesn't work with `yabai`
> Screen.Recording.2022-06-17.at.01.48.53.mov

I appear to have gotten this working by explicitly turning it on in `.yabai.rc`.

```yabai -m rule --add app="^Neovide$" manage=on```

then `neovide --frame none` works as expected

**Comment by sisrfeng** [comment]
neovide --frame none

beautiful!

![image](https://user-images.githubusercontent.com/53520949/195287400-2d27d0df-8069-493b-96dd-0c4d53fcc465.png)

**Comment by luxus** [comment]
> edit: it seems to be a little finnicky though, not 100% success rate

did you found a better way by now?

**Comment by monologconnor** [comment]
> 

same, the `--frame` option is not valid under Windows, the only two options available were `full` and `none` 😢

is there any possibility to implement the frameless feature in Windows? Just like the `adaptive dark` or `ayu` theme ([repo page](https://github.com/dempfi/ayu)) in Sublime Text?

![image](https://user-images.githubusercontent.com/38450642/216260677-5faf1d07-5474-48da-a04a-eab44bb83d05.png)

**Comment by fredizzimo** [comment]
Closing this, `--frame none` works in Windows as well with Neovide 0.11.0

**Comment by sisrfeng** [comment]
> working bar (not sure with the exact name) of my IME (sogou拼音输入法) disappear.

Still, `--frame none` makes the working bar (not sure with the exact name) of my IME (sogou拼音输入法) disappear, making it hard for Chinese input

**Comment by sisrfeng** [comment]
If `decorations: none`, Alacritty also has the problem

**Comment by fredizzimo** [comment]
@sisrfeng Could you create an issue in https://github.com/rust-windowing/winit, saying that both Alacritty and Neovide is broken?

**Comment by fredizzimo** [comment]
Sorry, I'm closing this again. You should open a new issue for `decorations: none and IME here as well, so that it's easier to track.
